pkg/sensors/exec/exec_test.go

```diff
@@ -945,23 +948,125 @@ func TestExecProcessCredentialsSuid(t *testing.T) {
+		WithUid(0).WithEuid(0).WithSuid(0).WithFsuid(0).
+		WithGid(0).WithEgid(0).WithSgid(0).WithFsgid((0))
+	procExecNoPrivsChanged := ec.NewProcessChecker().
+		WithBinary(sm.Full(testBin)).WithProcessCredentials(noCredsChange).WithBinaryProperties(nil)
+	execNoPrivsChangedChecker := ec.NewProcessExecChecker("exec").WithProcess(procExecNoPrivsChanged)
 	if err := exec.Command(testBin).Run(); err != nil {
 		t.Fatalf("Failed to execute '%s' binary: %s\n", testBin, err)
 	}
+	/* Executing a setuid and setgid to root but we are already running as root
+	 * so no privilege change should be detected, same filters as above apply.
+	 */
+	if err := os.Chown(testSuid, 0, 0); err != nil {
+		t.Fatalf("Chown() on '%s' binary error: %s\n", testSuid, err)
+	}
+	if err := os.Chmod(testSuid, 0754|os.ModeSetuid|os.ModeSetgid); err != nil {
+		t.Fatalf("Chown() on '%s' binary error: %s\n", testSuid, err)
+	}
+	procExecSetuidRootNoPrivsChanged := ec.NewProcessChecker().
+		WithBinary(sm.Full(testSuid)).WithProcessCredentials(noCredsChange).WithBinaryProperties(nil)
+	execSetuidRootNoPrivsChangedChecker := ec.NewProcessExecChecker("exec").WithProcess(procExecSetuidRootNoPrivsChanged)
+	if err := exec.Command(testSuid).Run(); err != nil {
+		t.Fatalf("Failed to execute '%s' binary: %s\n", testSuid, err)
+	}
+
+	checker := ec.NewUnorderedEventChecker(execNoPrivsChangedChecker, execSetuidRootNoPrivsChangedChecker)
+	err = jsonchecker.JsonTestCheck(t, checker)
+	assert.NoError(t, err)
+}
+
+// Test running with different combinations of setgid bit set
+//  1. setgid() systemcall to arbitrary gid value then exec binary to
+//     assert credentials did not change.
+//  2. executes a set-group-ID to root binary asserting that we detect
+//     the setgid bit set + the privileges changed due to the setgid bit
+//     being set to root group.
+//  3. executes a set-group-ID to non root, it is set to arbitrary value
+//     to assert that binary execution detects the setgid bit but we do
+//     not report as a privilege changed execution as the target group
+//     is not root.
+func TestExecProcessCredentialsSetgidChanges(t *testing.T) {
+	var doneWG, readyWG sync.WaitGroup
+	defer doneWG.Wait()
+
+	ctx, cancel := context.WithTimeout(context.Background(), tus.Conf().CmdWaitTime)
+	defer cancel()
+
+	obs, err := observertesthelper.GetDefaultObserver(t, ctx, tus.Conf().TetragonLib, observertesthelper.WithMyPid())
+	if err != nil {
+		t.Fatalf("Failed to run observer: %s", err)
+	}
+
+	testBin := testutils.RepoRootPath("contrib/tester-progs/nop")
+	// We should be able to create suid on local mount point
+	testSuid := testutils.RepoRootPath("contrib/tester-progs/suidnop")
+	if err := testutils.CopyFile(testSuid, testBin, 0754|os.ModeSetuid|os.ModeSetgid); err != nil {
+		t.Fatalf("Failed to copy binary: %s", err)
+	}
+
+	t.Cleanup(func() {
+		if err = syscall.Setuid(0); err != nil {
+			t.Fatalf("Failed to restore uid to 0 :  %s\n", err)
+		}
```

Comment:
Why are we setting setuid to `0` here? Shouldn't we set it to whatever the previous value was?

Reply:
ok got you! will fix it

Reply:
I did remove it, however we have been running tests with root for a long time, so if we ever change this these tests and some others may fail...